packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,198 @@
+import { Customer, PaymentSessionStatus } from "../models"
+import { MedusaContainer } from "../types/global"
+import {
+  PaymentProcessorContext,
+  PaymentProcessorError,
+  PaymentProcessorSessionResponse,
+} from "../types/payment-processor"
+
+/**
+ * The new payment service plugin interface
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
+
+  /**
+   * This method will be called once when the plugin will be registered
+   */
+  init(): Promise<void>
```

Comment:
thought: I don't think we need this method in the interface. At least not in the first iteration.

Reply:
Yes I can remove it, it was more to move the initialisation of the provider from the constructor to this method knowing that it can be async. I ll remove it

Reply:
@olivermrbl thoughts, the initialization of an external package can be async. Since the constructor is always since it is possible to call a method while the initialization is not finalized cause the promise is not resolved yet. wdyt?

Reply:
I am fine keeping it in. But then I suggest renaming it to be specific to the initialization of a given SDK/client/etc. and the current `init` could be mixed up with the initialization of a payment. Wdyt?

Reply:
`initialisePaymentProcessor` ? wdyt?

Reply:
Sorry, I misplaced my comment. To avoid blocking this PR further, I'm fine with just using `init`. My apologies for going back and forth on this one.